### PR TITLE
Visual Prolog programming language

### PR DESCRIPTION
**Pull request by tholp**
Added Visual Prolog language (https://www.visual-prolog.com).

### CODE REVIEW

File: pygments/lexers/vip.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    minorendkw = ('try', 'foreach', 'if')
+    minorkwexp = ('and', 'catch', 'do', 'else', 'elseif', 'erroneous', 'externally', 'failure', 'finally', 'foreach', 'if', 'or', 'orelse', 'otherwise', 'then',
+        'try', 'div', 'mod', 'rem', 'quot')
+    dockw = ('short', 'detail', 'end', 'withdomain')
+    tokens = {
+        'root': [
+            (r'\s+', Whitespace),
+            (words(minorendkw, prefix=r'\bend\s+', suffix=r'\b'), Keyword.Minor),
+            (r'end', Keyword),
+            (words(minorkwexp, suffix=r'\b'), Keyword.Minor),
```
[the remark this answers is not in the document]
Comment (tholp) [reply pygments/lexers/vip.py:18]: Yes

File: pygments/lexers/vip.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        'comment': [
+            include('commentdoc'),
+            (r'[^@*/]+', Comment),
+            (r'/\*', Comment, '#push'),
+            (r'\*/', Comment, '#pop'),
+            (r'[*/]', Comment),
+        ],
+        'stringescape' : [
+            (r'\\u[0-9a-fA-F]{4}', String.Escape),
+            (r'\\u[0-9a-fA-F]{0,3}', String.Escape.Error),
```
[the remark this answers is not in the document]
Comment (tholp) [reply pygments/lexers/vip.py:52]: It does mean an invalid escape sequence. I can remove that rule, if you think.

**Comment by tholp** [comment]
Common Lisp is not my language, but I believe that '^\s*(defun\s' will be a very good indicator for that language.  I can add the rule if you like.

**Comment by tholp** [comment]
Fixed

**Comment by tholp** [comment]
Yes, thank you for your patience :-).  I have successfully run 'py -m tox" now.

**Comment by Anteru** [comment]
Still failing CI. Can you try to run `tox -e check` locally first? Otherwise this will burn a few more trees before we get there :)

**Comment by tholp** [comment]
tox is not a stand alone command on my computer.  And it does not appear to have a -e option.  Running

py -m tox -- -c check

says:

...
................................................................................................................ [ 91%]
..............................................s................................................................. [ 94%]
.........................................................s..........................................             [ 96%]
tests\test_coffeescript.py ....................                                                                  [ 96%]
tests\test_html_formatter_linenos_elements.py .................................................................. [ 98%]
..............................................................                                                   [100%]

========================================== 4140 passed, 12 skipped in 19.35s ==========================================
.pkg: _exit> python C:\Users\puls\AppData\Roaming\Python\Python39\site-packages\pyproject_api\_backend.py True setuptools.build_meta
  py: OK (23.48=setup[3.48]+cmd[20.00] seconds)
  congratulations :) (23.73 seconds)

Maybe it is one of the 12 skipped tests that should have failed.

**Comment by tholp** [comment]
Generated new _mapping.py files.  Which unfortunately get <CR><LF> line endings (and have to be corrected manually afterwards).

Anyways, now the out put from py -m tox - e check is:

$ py -m tox -e check
.pkg: _optional_hooks> python C:\Users\puls\AppData\Roaming\Python\Python39\site-packages\pyproject_api\_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_editable> python C:\Users\puls\AppData\Roaming\Python\Python39\site-packages\pyproject_api\_backend.py True setuptools.build_meta
.pkg: build_editable> python C:\Users\puls\AppData\Roaming\Python\Python39\site-packages\pyproject_api\_backend.py True setuptools.build_meta
check: install_package> python -I -m pip install --force-reinstall --no-deps C:\_git-ext\pygments\src\.tox\.tmp\package\23\Pygments-2.15.1-0.editable-py3-none-any.whl
check: commands[0]> python scripts/check_crlf.py pygments external
check: commands[1]> python scripts/detect_missing_analyse_text.py --skip-no-aliases
check: commands[2]> flake8 --select F --ignore F401 pygments
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Users\puls\AppData\Local\Programs\Python\Python39\lib\threading.py", line 954, in _bootstrap_inner
    self.run()
  File "C:\Users\puls\AppData\Local\Programs\Python\Python39\lib\threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\puls\AppData\Local\Programs\Python\Python39\lib\multiprocessing\pool.py", line 519, in _handle_workers
    cls._wait_for_updates(current_sentinels, change_notifier)
  File "C:\Users\puls\AppData\Local\Programs\Python\Python39\lib\multiprocessing\pool.py", line 499, in _wait_for_updates
    wait(sentinels, timeout=timeout)
  File "C:\Users\puls\AppData\Local\Programs\Python\Python39\lib\multiprocessing\connection.py", line 884, in wait
    ready_handles = _exhaustive_wait(waithandle_to_obj.keys(), timeout)
  File "C:\Users\puls\AppData\Local\Programs\Python\Python39\lib\multiprocessing\connection.py", line 816, in _exhaustive_wait
    res = _winapi.WaitForMultipleObjects(L, False, timeout)
ValueError: need at most 63 handles, got a sequence of length 66

And those errors are meaningless to me.

**Comment by tholp** [comment]
The tox command leaves 70 python.exe processes running.

Not being able to figure out how to run the "check" I have instead added a doc string with copyright.

And will (sorry) waste a couple of your trees.

**Comment by tholp** [comment]
I was able to run 'py -m tox -e check' on a computer with fewer processors (I think the number of processors makes influences the number of python processes that are spawned and 64 processors is probably too many (there was something about too many handles in the errors).

However on that computer the -e check succeeds even though it failed when I pushed it to git.

Tried once more hope this time will succeed (I hoped that the other times as well).

**Comment by tholp** [comment]
I am blank about the permission thing.

Also my test run says no errors:

check: commands[2]> flake8 --select F --ignore F401 pygments
check: commands[3]> python scripts/check_sources.py -i pygments/lexers/_mapping.py -i docs/_build -i pygments/formatters/_mapping.py -i pygments/unistring.py -i tests/support/empty.py
**No errors found.**
check: commands[4]> python scripts/count_token_references.py --minfiles=1 --maxfiles=1 --minlines=1 --maxlines=3 --subtoken

Anyways I figured out that the underlining was wrong length, and have fixed that also.

**Comment by tholp** [comment]
And finally the testsuite succeeded :-).

**Comment by tholp** [comment]
Is there any problems I have to deal with?

**Comment by tholp** [comment]
Of, course.  Have a nice vacation.

**Pull request merged**